model-client/src/commonMain/kotlin/org/modelix/model/IncrementalBranch.kt

```diff
@@ -0,0 +1,488 @@
+        override val tree: ITree
+            get() = IncrementalTree(transaction.tree)
+    }
+
+    inner class IncrementalWriteTransaction(transaction: IWriteTransaction) : IncrementalTransaction<IWriteTransaction>(transaction), IWriteTransaction {
+        override var tree: ITree
+            get() = IncrementalTree(transaction.tree)
+            set(value) {
+                val oldTree = transaction.tree
+                transaction.tree = if (value is IncrementalTree) value.tree else value
+                createEventsForTreeChange(oldTree, value)
+            }
+        override fun setProperty(nodeId: Long, role: String, value: String?) {
+            transaction.setProperty(nodeId, role, value)
+            modified(PropertyDependency(this@IncrementalBranch, nodeId, role))
+        }
+
+        override fun setReferenceTarget(sourceId: Long, role: String, target: INodeReference?) {
+            transaction.setReferenceTarget(sourceId, role, target)
+            modified(ReferenceDependency(this@IncrementalBranch, sourceId, role))
+        }
+
+        override fun moveChild(newParentId: Long, newRole: String?, newIndex: Int, childId: Long) {
+            val oldParentId = transaction.getParent(childId)
+            val oldRole = transaction.getRole(childId)
+            transaction.moveChild(newParentId, newRole, newIndex, childId)
+            modified(ChildrenDependency(this@IncrementalBranch, oldParentId, oldRole))
+            modified(ChildrenDependency(this@IncrementalBranch, newParentId, newRole))
+            modified(ContainmentDependency(this@IncrementalBranch, childId))
+        }
+
+        override fun addNewChild(parentId: Long, role: String?, index: Int, concept: IConcept?): Long {
+            val childId = transaction.addNewChild(parentId, role, index, concept)
+            modified(ChildrenDependency(this@IncrementalBranch, parentId, role))
+            return childId
+        }
+
+        override fun addNewChild(parentId: Long, role: String?, index: Int, concept: IConceptReference?): Long {
+            val childId = transaction.addNewChild(parentId, role, index, concept)
+            modified(ChildrenDependency(this@IncrementalBranch, parentId, role))
+            return childId
+        }
+
+        override fun addNewChild(parentId: Long, role: String?, index: Int, childId: Long, concept: IConcept?) {
+            transaction.addNewChild(parentId, role, index, childId, concept)
+            modified(ChildrenDependency(this@IncrementalBranch, parentId, role))
+        }
+
+        override fun addNewChild(
+            parentId: Long,
+            role: String?,
+            index: Int,
+            childId: Long,
+            concept: IConceptReference?,
+        ) {
+            transaction.addNewChild(parentId, role, index, childId, concept)
+            modified(ChildrenDependency(this@IncrementalBranch, parentId, role))
```

Comment:
Do we not need a call to `modified(UnclassifiedNodeDependency(this@IncrementalBranch, nodeId))`, becauase of `.containsNode`?

Reply:
Yes, that was indeed incorrect.